### PR TITLE
Add method argument to SimulatedAnnealing.autofit

### LINKED ISSUE #117: Add linear updating method to SimulatedAnnealing.autofit
`SimulatedAnnealing.autofit` assumes that the final temperature always goes to 1:

https://github.com/N-Wouda/ALNS/blob/8ef4665c9cb27923497e11dccf3e75d9b58821bd/alns/accept/SimulatedAnnealing.py#L104-L128

It would be nice to have an `end_worse` argument for simulated annealing, so that we can have final temperatures other than 1. The current `worse` parameter could rephrased to `start_worse`. 
- Issue: I don't know if there is a good default value for `start_worse` to make this a non-breaking change

It would also be nice to have an `method` argument to choose which updating method should be used. The current implementation assumes exponential updating. This is definitely not breaking.

For reference, RRT already has the two above features:
https://github.com/N-Wouda/ALNS/blob/8ef4665c9cb27923497e11dccf3e75d9b58821bd/alns/accept/RecordToRecordTravel.py#L109-L167

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR:

- [x] Closes #117.
- [x] Adds and passes relevant tests.

### CODE REVIEW

File: alns/accept/SimulatedAnnealing.py [hunk 1]
```diff
@@ -55,6 +55,11 @@ def __init__(
                 "start_temperature < end_temperature not understood."
             )
 
+        method = method.lower()
```
Comment (leonlan) [inline alns/accept/SimulatedAnnealing.py:58]: Question: why do we want this? 
Suggestion: if we keep this, then we should also add this to RecordToRecordTravel.

File: alns/accept/SimulatedAnnealing.py [hunk 2]
```diff
@@ -154,12 +168,16 @@ def autofit(
         if not (0 < accept_prob < 1):
             raise ValueError("accept_prob outside (0, 1) not understood.")
 
-        if num_iters < 0:
+        if num_iters <= 0:
             raise ValueError("Negative number of iterations not understood.")
```
Comment (leonlan) [inline alns/accept/SimulatedAnnealing.py:172]: Rephrase: "Nonpositive number of iterations not understood."

File: alns/accept/SimulatedAnnealing.py (hunk 1 above)
Comment (N-Wouda) [reply alns/accept/SimulatedAnnealing.py:58]: I forgot we already have this, in `update.py` (which is called from almost everywhere). So I'm removing it here.

File: alns/accept/SimulatedAnnealing.py [hunk 3]
```diff
@@ -154,12 +163,16 @@ def autofit(
...
+        if num_iters <= 0:
+            raise ValueError("number of iterations <= 0 not understood.")
 
         start_temp = -worse * init_obj / np.log(accept_prob)
-        step = (1 / start_temp) ** (1 / num_iters)
+
+        if method == "exponential":
+            step = (1 / start_temp) ** (1 / num_iters)
+        else:
+            step = (start_temp - 1) / num_iters
```
Comment (leonlan) [inline alns/accept/SimulatedAnnealing.py:174]: Issue: removing the `method.lower()` from here does not lead to the correct behavior.  E.g., `method == "EXPONENTIAL"` will be interpreted as linear step.

Comment (leonlan) [reply alns/accept/SimulatedAnnealing.py:174]: One way to fix this is to check for `method.lower()`.

Comment (N-Wouda) [reply alns/accept/SimulatedAnnealing.py:174]: Yeah no, I got this wrong. Lemme fix.

**Comment by N-Wouda** [comment]
@leonlan this should be better

**Review by leonlan — APPROVED** [review]

**Pull request merged**